### ISSUE TITLE
Dependent Dropdowns

### ISSUE THREAD

**Issue by rajivseelam** [trebuchetty/Handsontable-select2-editor#14]
I have two select2 dropdowns and I want to change options the second dropdown depending on first dropdown.

For example,

        {
          data: 'country_id',
          editor: 'select2', 
          renderer: customDropdownRenderer,
          select2Options: { 
                  data: {!! $production_units !!} ,
                  dropdownAutoWidth: true,
          }
        },
        {
          data: 'state_id',
          editor: 'select2', 
          renderer: customDropdownRenderer,
          select2Options: { 
                  data: [],
                  dropdownAutoWidth: true,
                  width: 'resolve'
          }
        },
Depending on country_id, I want to change select2 options of state_id. I know how to make this work with just select2, but I am not able to figure out how to make it work with handsontable.

I have change select2Options in afterChange, but how to do that?

  afterChange: function (change, source) {

        if(change)
        {

          if(change[0][1] == 'country_id')
          {
            $.get("/api/states?country="+change[0][3], function(data){

                 //What should be done here?

            });
          }

        }

      },

I have posted a question at 
http://stackoverflow.com/questions/31691214/handsontable-select2-dependent-dropdowns

**Comment by trebuchetty** [comment]
Answer from Stackoverflow:
You retrieve all the states for that country. Make sure the states list is in the same format that select2 expects.

Then loop through the columns list and determine the dependent column for the current column that has been modified.

Get the cell properties:

var cellProperties = instance.getCellMeta(rowIndex, dependentColumnIndex);
Update the select2 source:

cellProperties['select2Options'].data = [states list];


I might have done this slightly differently, but this is probably better than the approach I had thought of.

**Comment by rajivseelam** [comment]
Hi,

Following the same example, this solution is not working.

Let's say I chose "Australia" in first row, it shows states in Australia in the states column of the corresponding row.

In another row below it, If i chose "India", then the problem starts, The states data of first row in which we selected "Australia" is also changes to states list of "India".

In short, select2Options are changing for all rows.
